src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -81,24 +81,42 @@ impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
             self.tag().get_str(),
             (key_locks.len() + key_old_writes.len()) as f64,
         );
-
+        // Check next key firstly.
+        let next_lock_key = if has_remain_locks {
+            key_locks.pop().map(|(key, _)| key)
+        } else {
+            None
+        };
+        let next_write_key = if has_remain_writes && !key_old_writes.is_empty() {
+            key_old_writes.pop().map(|(key, _)| key)
+        } else if has_remain_writes && key_old_writes.is_empty() {
+            // We haven't read any write yet, so we need to read the writes in the next
+            // batch later.
+            self.next_write_key
+        } else {
+            None
+        };
         if key_locks.is_empty() && key_old_writes.is_empty() {
+            if next_write_key.is_some() || next_lock_key.is_some() {
+                // When there are no keys that need to be flashbacked in this batch, but since
```

Comment:
```suggestion
                // Although there is no key left in this batch, keep processing the next batch since next key(write or lock) exist.
```